TypeCobol/Compiler/CompilationDocument.cs

```diff
@@ -476,6 +482,8 @@ public void UpdateTokensLines(System.Action onVersion = null)
                 if (scanAllDocumentLines)
                 {
                     ScannerStep.ScanDocument(TextSourceInfo, compilationDocumentLines, CompilerOptions, CopyTextNamesVariations, initialScanStateForCopy);
+                    //Notify any Language Server listener that the whole document has been rescanned
+                    this.LanguageServer?.UpdateTokensLines(null, this);
```

Comment:
There is already a listener at the end of this method:
tokensLinesChanged
Can you use to avoid to change CompilationDocument?

Reply:
CanI call the listener if the entire document has been changed, by passing null as changing?

Reply:
Can you create a second generic listerner in this case ?
I think it's better to have a generic behavior that we can reuse for different purpose.

Reply:
Yes I can.

Reply:
Or yes maybe passing null or create a DocumentChangedEvent<ITokensLine>